### PR TITLE
Modify confusing deprecation comment for generic top level prefix proxy.

### PR DESCRIPTION
**Pull request by xiangpengzhao**
**What this PR does / why we need it**:
The current comment is confusing.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
none
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/installer.go [hunk 1]
```diff
@@ -427,10 +427,8 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 		actions = appendIf(actions, action{"DELETE", itemPath, nameParams, namer, false}, isDeleter)
 		actions = appendIf(actions, action{"WATCH", "watch/" + itemPath, nameParams, namer, false}, isWatcher)
 		// We add "proxy" subresource to remove the need for the generic top level prefix proxy.
-		// The generic top level prefix proxy is deprecated in v1.2, and will be removed in 1.3, or 1.4 at the latest.
-		// TODO: DEPRECATED in v1.2.
+		// The generic top level prefix proxy is deprecated in v1.2, and should be removed in v2 API.
```
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/endpoints/installer.go]: This updated comment isn't correct.  The original is correct.  The top level proxy should be removed

**Comment by xiangpengzhao** [comment]
I once tried to remove it but some tests failed. Seems like something still depends on it. I will reproduce it.

File: staging/src/k8s.io/apiserver/pkg/endpoints/installer.go (hunk 1 above)
Comment (xiangpengzhao) [reply staging/src/k8s.io/apiserver/pkg/endpoints/installer.go]: I once tried to remove it but some tests failed. Seems like something still depends on it. I will reproduce it.

**Comment by xiangpengzhao** [comment]
@deads2k I found some discussions : https://github.com/kubernetes/kubernetes/pull/30259#issuecomment-238636391
As a testing, I remove the code in a separate commit to see if the CI will be happy :)

/cc @nikhiljindal

**Comment by xiangpengzhao** [comment]
kindly ping @nikhiljindal for confirmation :)

**Comment by xiangpengzhao** [comment]
/close
in favor of https://github.com/kubernetes/kubernetes/pull/59884

**Pull request closed without merging**